fivetran_provider_async/sensors.py

```diff
@@ -114,12 +170,41 @@ def hook(self) -> FivetranHook:
             retry_delay=self.fivetran_retry_delay,
         )
 
-    def poke(self, context: Context):
-        if self.previous_completed_at is None:
-            self.previous_completed_at = self.hook.get_last_sync(self.connector_id, self.xcom)
+    @property
+    def xcom(self) -> str:
+        import warnings
 
-        return self.hook.get_sync_status(
-            self.connector_id, self.previous_completed_at, self.reschedule_wait_time
+        warnings.warn(
+            "`xcom` attribute is deprecated. Use `completed_after_time` attribute instead.",
+            DeprecationWarning,
+            stacklevel=2,
+        )
+        return self.completed_after_time
+
+    @xcom.setter
+    def xcom(self, value: str) -> None:
+        import warnings
+
+        warnings.warn(
+            "`xcom` attribute is deprecated. Use `completed_after_time` attribute instead.",
+            DeprecationWarning,
+            stacklevel=2,
+        )
+        self.completed_after_time = value
+
+    def poke(self, context: Context):
+        if self._completed_after_time_rendered is None:
+            if not self.completed_after_time:
+                self._completed_after_time_rendered = self.hook.get_last_sync(self.connector_id)
+            else:
+                self._completed_after_time_rendered = timezone.parse(self.completed_after_time)
```

Comment:
I did a quick run on the old example but it failed here 
Example: https://github.com/astronomer/airflow-provider-fivetran-async/blob/main/fivetran_provider_async/example_dags/example_fivetran_async.py#L33
Log
```
Traceback (most recent call last):
  File "/usr/local/lib/python3.11/site-packages/pendulum/parsing/__init__.py", line 131, in _parse
    dt = parser.parse(
         ^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/dateutil/parser/_parser.py", line 1368, in parse
    return DEFAULTPARSER.parse(timestr, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/dateutil/parser/_parser.py", line 643, in parse
    raise ParserError("Unknown string format: %s", timestr)
dateutil.parser._parser.ParserError: Unknown string format: None
During handling of the above exception, another exception occurred:
Traceback (most recent call last):
  File "/usr/local/airflow/airflow_provider_fivetran_async/fivetran_provider_async/sensors.py", line 149, in execute
    elif not self.poke(context):
             ^^^^^^^^^^^^^^^^^^
  File "/usr/local/airflow/airflow_provider_fivetran_async/fivetran_provider_async/sensors.py", line 200, in poke
    self._completed_after_time_rendered = timezone.parse(self.completed_after_time)
                                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/airflow/utils/timezone.py", line 207, in parse
    return pendulum.parse(string, tz=timezone or TIMEZONE, strict=strict)  # type: ignore
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/pendulum/parser.py", line 29, in parse
    return _parse(text, **options)
           ^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/pendulum/parser.py", line 45, in _parse
    parsed = base_parse(text, **options)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/pendulum/parsing/__init__.py", line 74, in parse
    return _normalize(_parse(text, **_options), **_options)
                      ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/pendulum/parsing/__init__.py", line 135, in _parse
    raise ParserError("Invalid date string: {}".format(text))
pendulum.parsing.exceptions.ParserError: Invalid date string: None
[2023-10-17, 07:27:15 UTC] {taskinstance.py:1398} INFO - Marking task as FAILED. dag_id=example_fivetran_async, task_id=fivetran_async_sensor, execution_date=20231017T072549, start_date=20231017T072715, end_date=20231017T072715
[2023-10-17, 07:27:15 UTC] {standard_task_runner.py:104} ERROR - Failed to execute job 414 for task fivetran_async_sensor (Invalid date string: None; 324)
[2023-10-17, 07:27:15 UTC] {local_task_job_runner.py:228} INFO - Task exited with return code 1
[2023-10-17, 07:27:15 UTC] {taskinstance.py:2776} INFO - 0 downstream tasks scheduled from follow-on schedule check
```
Is this expected?

Reply:
Sorry for delays, I'd had a very busy last few weeks.

I started a new job recently and don't have Airflow + Fivetran set up just yet (will be setting it up soon).

That said, I think I can see what happened here.

---

The error message shows the following:

```
Invalid date string: None
```

So it looks like what happened is the `FivetranOperator` returned `None`. Looking at the code, we see (simplified):

```python
    def execute(self, context: Context) -> None | str:
        ...

        if not self.wait_for_completion:
            return last_sync

        if not self.deferrable:
            self._wait_synchronously(pendulum.parse(last_sync))  # type: ignore[arg-type]
            return None
        else:
            ...
            return None
```

So it only returns `last_sync` (str) when when `wait_for_completion` is `False`. This makes sense, since in this case you'd be implementing a sort of async-await pattern with operator+sensor and the timestamp is used as the "future" so to speak.

The issue therefore is that the example DAG needs to show `wait_for_completion=False`.

---

Reading these examples, I actually found them a little confusing.

I now wrote them a little bit differently to clarify the new behaviors:

- `example_fivetran.py` shows a sync example. It is unnecessary to use a sensor now by default, so I removed the sensor.
- `example_fivetran_async.py` also does not need to use the sensor, since the `FivetranOperator` awaits the Fivetran sync (both in sync and async moe).
- `example_fivetran_xcom.py` needed `wait_for_completion=False` to be set for that example to make any sense, which I didn't have.

That said, there are still some possible issues with the Bigquery and Bqml examples? Since they both uses the unnecessary `FivetranSensor`. (It is unnecessary because the `FivetranOperator` does the job of waiting). Maybe we remove the `FivetranSensor`s from there too?

We can also potentially provide an example of when Fivetran is being externally scheduled and the `{{ data_interval_end }}` is being used as the completed time?